.github/workflows/pipeline.yml

```diff
@@ -13,11 +13,11 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up JDK 17
-        uses: actions/setup-java@v3
+      - uses: actions/checkout@v4
+      - name: Set up JDK 23
+        uses: actions/setup-java@v4
         with:
-          java-version: '17'
+          java-version: '23'
```

Comment:
We've chosen to stick with Java 17 until its end of premier support date (2026-09-30) in order to allow the largest possible range of up-to-date JDK users to use this project's code unmodified.

Reply:
that makes sense

Reply:
i also changed it back in the pom in that case